sway/input/text_input.c

```diff
@@ -312,15 +312,17 @@ static void input_popup_update(struct sway_input_popup *popup) {
 	}
 
 	if (!popup->popup_surface->surface->mapped) {
+		popup->scene_tree = NULL;
+		popup->desc.view = NULL;
 		return;
 	}
 
-	if (popup->scene_tree != NULL) {
-		wlr_scene_node_destroy(&popup->scene_tree->node);
-		if (popup->desc.relative != NULL) {
-			wlr_scene_node_destroy(popup->desc.relative);
-		}
+	wlr_scene_node_destroy(&popup->scene_tree->node);
+	if (popup->desc.relative != NULL) {
```

Comment:
Not yet, this if statement can be dropped and all of this can be move above the mapped check.

Reply:
ok, I think now it is done

Reply:
Thank you very much! The bug I reported was fixed and now Sway doesn't crash on input in wofi window.
Now another small bug was found. The position of the popup is not fixed inside the screen if the wofi window is on the bottom of the screen.
Simplified config is below.
https://gist.github.com/hikyae/e67119cfafb0a4ec76b1a6ba840cb861
[recording_2024_02_15_23_33_09.webm](https://github.com/swaywm/sway/assets/144245294/ba8b28aa-c393-40ce-8994-7bdaafcc6e08)

Reply:
Emm, you mean wofi window can be on the bottom of the screen? oh do do it @hikyae

Reply:
lines=1
y=1005
in ~/.config/wifi/config
https://gist.github.com/hikyae/e67119cfafb0a4ec76b1a6ba840cb861

Reply:
@hikyae try the new pr now, it should be fixed

Reply:
@Decodetalkers 
I've tried the latest sway-im-git now.
But it's not fixed yet. The popup still appears out of the screen.

Reply:
Can you make a video for me? @hikyae

Reply:
@Decodetalkers 
Have you watched this?
https://github.com/swaywm/sway/assets/144245294/ba8b28aa-c393-40ce-8994-7bdaafcc6e08
Now the symptom remains. Or do you want another video? What do you want to watch?

Reply:
oh, I see, when it is bottom. you think the popup should be from top

Reply:
Yes

Reply:
@hikyae fixed, it was a mistake

Reply:
@Decodetalkers 
It works!
Thank you very much!